llvm/lib/SYCLLowerIR/CompileTimePropertiesPass.cpp

```diff
@@ -270,11 +271,23 @@ attributeToExecModeMetadata(const Attribute &Attr, Function &F) {
     return std::pair<std::string, MDNode *>("ip_interface",
                                             getIpInterface("csr", Ctx, Attr));
 
-  if (AttrKindStr == SYCL_REGISTER_ALLOC_MODE_ATTR &&
```

Comment:
Might be easier for future maintenance if we can split this into two if-blocks - one for SYCL_REGISTER_ALLOC_MODE_ATTR and one for SYCL_GRF_SIZE_ATTR.
Please also add a TODO comment to remind of removing SYCL_REGISTER_ALLOC_MODE_ATTR when ready.

Thanks

Reply:
thanks, i added the TODO

if we added two blocks, we would see code duplication with creating the `RegisterAllocMode` MD, and we won't be extending the `sycl::detail` version as it is deprecated, so i don't think it should a big issue.